authors/apps/authentication/backends.py

```diff
@@ -1,10 +1,45 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+import jwt
 
-"""Configure JWT Here"""
+from django.conf import settings
 
+from rest_framework import authentication, exceptions
+
+from .models import User
+
+class JWTAuthentication(authentication.BaseAuthentication):
+    authentication_header_prefix = 'token'
+
+    def authenticate(self, request):
+        """This method is called on every endpoint."""
+        request.user = None
+        auth_header = authentication.get_authorization_header(request).split()
+        auth_header_prefix = self.authentication_header_prefix.lower()
+        if not auth_header:
+            return None
```

Comment:
Raise `AuthenticationFailed` exceptions with unique messages for each of these header related mistakes.